slider/position.py

```diff
@@ -16,3 +16,21 @@ class Position(namedtuple('Position', 'x y')):
     The visible region of the osu! standard playfield is [0, 512] by [0, 384].
     Positions may fall outside of this range for slider curve control points.
     """
+
+class Point(namedtuple('Point', 'x y offset')):
```

Comment:
I think that I might go with `Position` and `TimedPosition`. While these are very verbose, they are unambiguous. I think I will change the `TimedPosition`. to be:
```python
class TimedPosition:
    __slots__ = 'position', 'time'

    def __init__(self, position, time):
        self.position = position
        self.time = time

    @property
    def x(self):
        return self.position.x

    @property
    def y(self):
        return self.position.y

    def __getitem__(self, ix):
        if ix == 0:
            return self.x
        elif ix == 1:
            return self.y
        elif ix == 2:
            return self.time
        raise IndexError(f'index {ix} is out of bounds, must be in range [0, 3)')

    def __iter__(self):
        yield self.x
        yield self.y
        yield self.time
```
This will avoid copying the x and y out of the position so much. and allow us to recover the position quickly at a slight overhead when getting just the x or y.